secrets.go

```diff
@@ -0,0 +1,132 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/buildkite/go-pipeline/ordered"
+	"gopkg.in/yaml.v3"
+)
+
+var _ interface {
+	json.Unmarshaler
+	ordered.Unmarshaler
```

Comment:
Because `Secrets` has a `MarshalYAML` method and we want some assurance that the YAML package will find and use it, we can add `yaml.Marshaler` here:

```suggestion
	ordered.Unmarshaler
	yaml.Marshaler
```

(This _should_ work even though the method takes a `Secrets` as its receiver, and this var is testing interface compatibility with a value of `*Secrets` rather than `Secrets`, because given a non-nil pointer to some data, Go is usually smart enough to dereference it in order to call a method on the non-pointer.)